ai-ml/jupyterhub/addons.tf

```diff
@@ -0,0 +1,244 @@
+  #---------------------------------------
+  enable_karpenter                  = true
+  karpenter_enable_spot_termination = true
+  karpenter_node = {
+    create_iam_role          = true
+    iam_role_use_name_prefix = false
+    # We are defining role name so that we can add this to aws-auth during EKS Cluster creation
+    iam_role_name = local.karpenter_iam_role_name
+  }
+
+  karpenter = {
+    timeout             = "300"
+    repository_username = data.aws_ecrpublic_authorization_token.token.user_name
+    repository_password = data.aws_ecrpublic_authorization_token.token.password
+  }
+
+}
+
+#---------------------------------------------------------------
+# IRSA for EBS
+#---------------------------------------------------------------
+module "ebs_csi_driver_irsa" {
+  source                = "terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks"
+  version               = "~> 5.14"
+  role_name             = format("%s-%s", local.name, "ebs-csi-driver")
+  attach_ebs_csi_policy = true
+  oidc_providers = {
+    main = {
+      provider_arn               = module.eks.oidc_provider_arn
+      namespace_service_accounts = ["kube-system:ebs-csi-controller-sa"]
+    }
+  }
+  tags = local.tags
+}
+
+#---------------------------------------------------------------
+# IRSA for VPC CNI
+#---------------------------------------------------------------
+module "vpc_cni_irsa" {
+  source                = "terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks"
+  version               = "~> 5.14"
+  role_name             = format("%s-%s", local.name, "vpc-cni")
+  attach_vpc_cni_policy = true
+  vpc_cni_enable_ipv4   = true
+  oidc_providers = {
+    main = {
+      provider_arn               = module.eks.oidc_provider_arn
+      namespace_service_accounts = ["kube-system:aws-node"]
+    }
+  }
+  tags = local.tags
+}
+#---------------------------------------------------------------
+# VPC CNI Addon should run before the nodes are created
+# Ideally VPC CNI with custom configuration values should be deployed before the nodes are created to use the correct VPC CNI config
+#---------------------------------------------------------------
+resource "aws_eks_addon" "vpc_cni" {
+  cluster_name             = module.eks.cluster_name
+  addon_name               = "vpc-cni"
+  addon_version            = data.aws_eks_addon_version.this.version
+  resolve_conflicts        = "OVERWRITE"
+  preserve                 = true # Ensure VPC CNI is not deleted before the add-ons and nodes are deleted during the cleanup/destroy.
+  service_account_role_arn = module.vpc_cni_irsa.iam_role_arn
+}
+
+#---------------------------------------------------------------
+# Data on EKS Kubernetes Addons
+#---------------------------------------------------------------
+# NOTE: This module will be moved to a dedicated repo and the source will be changed accordingly.
+module "kubernetes_data_addons" {
```

Comment:
this can be removed - you can set ad hoc helm charts in the blueprints addons module like this https://github.com/aws-ia/terraform-aws-eks-blueprints-addons/blob/c5f51610e469e741eceb1c4cb185d317cbe6ddd2/tests/complete/main.tf#L171-L202

Reply:
Idea is to publish this `terraform-aws-eks-data-addons`  as a new repo and build E2E tests for future releases. Some of these add-ons will be reused in other blueprints as well. We can simply update the blueprints with the tag for future versions without specifying the helm Chart version in the blueprint